lib/oidcstrategy.js

```diff
@@ -144,20 +146,26 @@ function onProfileLoaded(strategy, args) {
  *   - `responseMode`       For login only flows we should have token passed back to us in a POST
  *   - `validateIssuer`     if you have validation on, you cannot have users from multiple tenants sign in
  *   - `passReqToCallback`  if you want the Req to go back to the calling function for other processing use this.
- *
+ *   - `stateMaxAmount`     the max amount of state you want to save in session, default is 200
+ *   - `stateMaxAge`        the lifetime of state in session, default is 1800s = 30 minutes
+ *   - `nonceMaxAmount`     the max amount of nonce you want to save in session, default is 200
+ *   - `nonce`              the lifetime of nonce in session, default is 1800s = 30 minutes
  * Examples:
  *
  * passport.use(new OIDCStrategy({
  *   callbackURL: config.creds.returnURL,
- *   realm: config.creds.realm,
  *   scope: config.creds.scopes,
  *   clientID: config.creds.clientID,
  *   identityMetadata: config.creds.identityMetadata,
  *   responseType: config.creds.responseType,
  *   responseMode: config.creds.responseMode,
  *   validateIssuer: config.creds.validateIssuer,
  *   passReqToCallback: config.creds.passReqToCallback,
- *   loggingLevel: config.creds.loggingLevel
+ *   loggingLevel: config.creds.loggingLevel,
+ *   stateMaxAmount: config.creds.stateMaxAmount,
```

Comment:
OK I will change the name to 'nonceMaxAmount'  and 'nonceLifeTime'.

Yes, if we hit the max amount, we just kick out the oldest one even though it is not expired. In reality we have more than one nonce/state when a user opens the microsoft login page (where they can put password) on multiple tabs without putting in their password. The default number 10 should be a reasonable amount.